arcaflow_plugin_pcp/pcp_schema.py

```diff
@@ -21,81 +32,28 @@ class PcpInputParams:
             "Timeout in seconds after which to cancel the pmlogger collection"
         ),
     ] = None
-
-
-@dataclass
-class IntervalOutput:
-    interval: typing.Annotated[
-        int,
-        schema.id("@interval"),
-        schema.name("Interval ID"),
-        schema.description("The interval ID as reported by PCP"),
-    ]
-    timestamp: typing.Annotated[
-        str,
-        schema.id("@timestamp"),
-        schema.name("Interval timestamp"),
-        schema.description("The timestamp of the reported interval"),
-    ]
-    commit: typing.Annotated[
-        typing.Optional[typing.Dict[str, typing.Any]],
-        schema.name("commit"),
-        schema.description("The commit for the interval"),
-    ] = None
-    disk: typing.Annotated[
-        typing.Optional[typing.Dict[str, typing.Any]],
-        schema.name("disk"),
-        schema.description("The disk structure for the interval"),
-    ] = None
-    kbin: typing.Annotated[
-        typing.Optional[typing.Dict[str, typing.Any]],
-        schema.name("kbin"),
-        schema.description("The KB in value for the interval"),
-    ] = None
-    kbout: typing.Annotated[
-        typing.Optional[typing.Dict[str, typing.Any]],
-        schema.name("kbout"),
-        schema.description("The KB out value for the interval"),
-    ] = None
-    kernel: typing.Annotated[
-        typing.Optional[typing.Dict[str, typing.Any]],
-        schema.name("kernel"),
-        schema.description("The kernel CPU structure for the interval"),
-    ] = None
-    mem: typing.Annotated[
-        typing.Optional[typing.Dict[str, typing.Any]],
-        schema.name("mem"),
-        schema.description("The memory structure for the interval"),
-    ] = None
-    memused: typing.Annotated[
-        typing.Optional[typing.Dict[str, typing.Any]],
-        schema.name("memused"),
-        schema.description("The memory used value for the interval"),
-    ] = None
-    pktin: typing.Annotated[
-        typing.Optional[typing.Dict[str, typing.Any]],
-        schema.name("pktin"),
-        schema.description("The packets in for the interval"),
-    ] = None
-    pktout: typing.Annotated[
-        typing.Optional[typing.Dict[str, typing.Any]],
-        schema.name("pktout"),
-        schema.description("The packets out for the interval"),
+    pmlogger_conf: typing.Annotated[
+        typing.Optional[str],
+        schema.name("pmlogger configuration file"),
+        schema.description(
+            "Complete configuration file content for pmlogger as a multi-line string."
+            " If no config file is provided, a default one will be generated."
+        ),
```

Comment:
So you think that it's worth removing input validation by switching it to a raw config?

Reply:
This is to allow user flexibility and is optional. The input configuration is far too complex to reasonably build a schema for, and a "power" PCP user will likely have an existing config file that they want to use.

Reply:
@dustinblack has a good point

Reply:
In my opinion, the best route would be to have two separate steps, or one step with two options (probably with a discriminator field):
- Input the raw config
- Build the config from values with a full schema

This would make it so that if the user has a config, great! Use the config. If the user doesn't have a config, they get our nice schema to help. That's especially helpful if configuring the schema in the future potential UI.

Reply:
This may be possible. While the list of configuration parameters is very long, it may at least be static. I'll check into it.

Reply:
Looking into this further, I just don't think it is an option to provide a schema for the configuration file. The file supports extensibility for user definitions, macros, and conditionals.

For this plugin, the default condition is to create a generic configuration file. The option to provide a pre-formatted file as text is something I consider to be a "power user" feature and should perhaps be documented with some "at your own risk" language.

Reply:
One small concern that I have is the case where a person doesn't have PCP running locally, but needs a config to start with. It may be a good idea to somehow expose the config, or mention how to get it.

One option I see to make this possible include making a step to generate the config, and making it required. That complicates using the default step, which may make this solution not worth it, but it makes it more explicit.
The other option I see is to include the config used in the output.

Reply:
> One small concern that I have is the case where a person doesn't have PCP running locally, but needs a config to start with. It may be a good idea to somehow expose the config, or mention how to get it.

I don't quite understand this concern. In what condition would the case of PCP running locally be a use case for this plugin? All of the services required are deployed and started in the container (granted, the design here _does_ assume the plugin is used within a container and not as python directly). Providing the config file as a string is only optional, and if it is not provided, a default config file will be generated and used by pmlogger.

Reply:
My concern isn't running it locally. It's retrieving the default config. Is that documented or exported somewhere?

Reply:
I'm still not following the need for a user to retrieve the default config, but I did update the readme to describe how that config is generated. The default config is transient as part of the container run and is transparent to the user.